### PR TITLE
Exometer

### PR DESCRIPTION
**Pull request by michalwski**
Replaces folsom with exometer.

Goes on top of #354

### CODE REVIEW

File: apps/ejabberd/src/ejabberd_app.erl [hunk 1]
```diff
@@ -238,3 +240,10 @@ load_drivers([Driver | Rest]) ->
                           [erl_ddll:format_error(Reason)]),
             exit({driver_loading_failed, Driver, Reason})
     end.
+
+init_metrics() ->
```
Comment (ppikula) [inline apps/ejabberd/src/ejabberd_app.erl]: I would move this function to the mongoose_metrics module to be consistent with `mongoose_metrics:remove_all_metrics()`

**Comment by ppikula** [comment]
Can you add generic functions for histograms ?

**Comment by michalwski** [comment]
Histograms are planned for next PR where I (or somebody else) will add metrics measuring communication time with databases.

File: doc/operation-and-maintenance/Mongoose-metrics.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+| disco_info | spiral | host, hook | |
+| disco_local_features | spiral | host, hook | |
+| disco_local_identity | spiral | host, hook | |
+| disco_local_items | spiral | host, hook | |
+| disco_sm_features | spiral | host, hook | |
+| disco_sm_identity | spiral | host, hook | |
+| disco_sm_items | spiral | host, hook | |
+| host_config_update | spiral | host, hook | |
+| local_send_to_resource_hook | spiral | host, hook | |
+| mam_lookup_messages | spiral | host, hook | |
```
Comment (michalwski) [inline doc/operation-and-maintenance/Mongoose-metrics.md]: @arcusfelis, could you please describe MAM related metrics?

File: doc/operation-and-maintenance/Mongoose-metrics.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+| ----------- | ---- | ----- | ----------- |
+| nodeSessionCount | value | global, XMPP | Number of connected users on given MongooseIM's node|
+| totalSessionCount | value | global, XMPP | Total number of connected users to MongooseIM cluster|
+| uniqueSessionCount | value | global, XMPP | Number of unique users connected to MongooseIM cluster |
+| adhoc_local_commands | spiral | host, hook | Hook run when adhoc command was sent to the server's host |
+| adhoc_local_items | spiral | host, hook | Hook run when adhoc command was sent to a user|
+| anonymous_purge_hook | spiral | host, hook | Hook run after anonymous user disconnected |
+| c2s_stream_features | spiral | host, hook | Hook run to collect stream features from optional modules / extensions |
+| c2s_unauthenticated_iq | spiral | host, hook | Hook run when user sent iq before authentication |
+| disco_info | spiral | host, hook | |
```
Comment (michalwski) [inline doc/operation-and-maintenance/Mongoose-metrics.md]: Need some help with disco related hooks. Who was dealing with this recently and know about these hooks?

**Pull request merged**